daemons/based/based_callbacks.c

```diff
@@ -935,8 +935,7 @@ cib_process_request(xmlNode *request, bool privileged,
         if (op_reply == NULL && (needs_reply || local_notify)) {
             crm_err("Unexpected NULL reply to message");
             crm_log_xml_err(request, "null reply");
-            needs_reply = false;
-            local_notify = false;
+            goto done;
```

Comment:
This also skips the "Completed pre-sync update" trace log if `is_update` is true. I suspect that you don't care about that; just highlighting that the flags do a LITTLE bit more than tell us to skip to the end.

Reply:
Yes that's true.  I noticed that when I was re-checking these patches, but didn't go back and modify the commit message.  I don't think I want the "Completed pre-sync update..." log message in this case since it's a failure and not really a completion (I guess we could argue about that, but whatever).  I've updated the commit message.